Support read/write from/to local node filesystem (#29565)
ray-project/ray: doc/source/data/creating-datasets.rst

    :start-after: __read_parquet_az_begin__
    :end-before: __read_parquet_az_end__

Reading from Local Storage
==========================

In Ray Datasets, users often read from remote storage systems as described above. In

> some use cases users may want to read from local storage. There are three ways to read

Reviewer: ```suggestion
some use cases, users may want to read from local storage. There are three ways to read
```

> from local filesystem:

Reviewer: ```suggestion
from a local filesystem:
```

> * Providing a local filesystem path: For example, in ``ray.data.read_csv("my_file.csv")``,

Reviewer: ```suggestion
* **Providing a local filesystem path:** For example, in ``ray.data.read_csv("my_file.csv")``,
```

>   the given path will be resolved as a local filesystem path. Note: if the file exists
>   only on the local node and you run this read operation in distributed cluster, this
>   will fail as it cannot access the file from remote node.

Reviewer: Nit: Use note directive.
```suggestion
  the given path will be resolved as a local filesystem path.
  .. note:: If the file exists only on the local node and you run this read
    operation in a distributed cluster, this will fail as it cannot access the
    file from a remote node.
```

> * Using ``local://`` custom URI scheme: Similarly, this will be resolved to local

Reviewer: ```suggestion
* **Using ``local://`` custom URI scheme:** Similarly, this will be resolved to a local
```

  filesystem, e.g. ``ray.data.read_csv("local://my_file.csv")`` will read the
  same file as the approach above. The difference is that this scheme will ensure

>   all read tasks happen on the local node, so it's safe to run in a distributed

Reviewer: ```suggestion
  that all read tasks happen on the local node, so it's safe to run in a distributed
```

  cluster.

> * Using ``example://`` custom URI scheme: The paths with this scheme will be resolved

Reviewer: ```suggestion
* **Using ``example://`` custom URI scheme:** The paths with this scheme will be resolved
```

>   to ``python/ray/data/examples/data`` directory in Ray codebase. This scheme is used

Reviewer: Nit: This will technically resolve to the `ray/data/examples/data` path in the Ray package, e.g. `~/.local/lib/python3.7/site-packages/ray/data/examples/data`.
```suggestion
  to ``ray/data/examples/data`` directory in the Ray package. This scheme is used
```

    :start-after: __read_parquet_az_begin__
    :end-before: __read_parquet_az_end__

> Reading from Local Storage

Reviewer 2: Should also mention this works for writes, which is pretty useful.
